### ISSUE TITLE
java.io.IOException: Cannot run program "powershell.exe": CreateProcess error=2, The system cannot find the file specified

### ISSUE THREAD

**Issue by xshi0001** [sbt/sbt#5386]
## steps
sbt version: *1.3.4*  window10

## problem

```
xshi0@LAPTOP-L904TFAI MINGW64 /d/WorkPlace/Flink/2-demo/springBoot-flink
$ sbt
[warn] No sbt.version set in project/build.properties, base directory: D:\WorkPlace\Flink\2-demo\springBoot-flink
[error] java.io.IOException: Cannot run program "powershell.exe": CreateProcess error=2, 系统找不到指定的文件。
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? l
```

## expectation
how to solve？

**Comment by eed3si9n** [comment]
@xshi0001 Thanks for the report.

Likely the problem is coming from Coursier calling directories-jvm, which uses powershell to query for the AppLocal directory on Windows. First thing to try is to use the latest version of sbt, which is sbt 1.3.7 because there was a recent fix around this.

If that doesn't work, you can specify `COURSIER_HOME` environment variable so that it won't query for AppLocal directory to figure out the coursier directories. /cc @alexarchambault, @soc

**Comment by eed3si9n** [comment]
@alexarchambault, @soc Scala team is now hitting this problem - https://github.com/scala/scala-dev/issues/666

> okay, now at https://scala-ci.typesafe.com/view/scala-2.12.x/job/scala-2.12.x-integrate-windows/1644/console we see:
> 
> ```
> [error] Caused by: java.io.IOException: Cannot run program "powershell.exe": CreateProcess error=2, The system cannot find the file specified
> [error] 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.runCommands(Util.java:176)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.getWinDirs(Util.java:147)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.Util.getWinDirs(Util.java:120)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.ProjectDirectories.fromPath(ProjectDirectories.java:221)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.shaded.io.github.soc.directories.ProjectDirectories.from(ProjectDirectories.java:274)
> [error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.coursierDirectories(CoursierPaths.java:82)
> [error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.computeCacheDirectory(CoursierPaths.java:44)
> [error] 	at lmcoursier.internal.shaded.coursier.paths.CoursierPaths.cacheDirectory(CoursierPaths.java:70)
> [error] 	at lmcoursier.internal.shaded.coursier.paths.CachePath.defaultCacheDirectory(CachePath.java:93)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.CacheDefaults$.location$lzycompute(CacheDefaults.scala:16)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.CacheDefaults$.location(CacheDefaults.scala:16)
> [error] 	at lmcoursier.internal.shaded.coursier.cache.FileCache$.apply(FileCache.scala:1088)
> [error] 	at lmcoursier.CoursierDependencyResolution.update(CoursierDependencyResolution.scala:134)
> [error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:60)
> ```

**Comment by SyraLessThanThree** [comment]
powershell.exe has been renamed to pwsh.exe since switching to powershell core
i dont't even have a powershell.exe but i have a pwsh.exe it should provide the same functionallity

**Comment by eatkins** [comment]
@Paukler this is a problem with an upstream library: https://github.com/dirs-dev/directories-jvm/issues/29

**Comment by soc** [comment]
These seem to be two unrelated problems – if we want/need to rename the executable directories-jvm tries to call, this would be a new issue/PR.

**Comment by eatkins** [comment]
I would say that it's splitting hairs to call these problems unrelated. The original issue was that powershell.exe wasn't on the path even though it was on the system. In @Paukler's case the issue is that powershell.exe isn't even on the system but equivalent functionality is provided by pwsh.exe. Nevertheless, both the error message and the user experience (sbt crashes on startup) are the same in either case. If you look at my suggestion, https://github.com/dirs-dev/directories-jvm/issues/29#issuecomment-583896399, as to how to resolve the issue by trying different executable locations if powershell.exe is not on the path, it would be trivial to try pwsh.exe if powershell.exe is not found.

**Comment by soc** [comment]
I misread, not sure how https://github.com/dirs-dev/directories-jvm/issues/29#issuecomment-583913762 ended up on that issue...

**Comment by eatkins** [comment]
See https://github.com/dirs-dev/directories-jvm/pull/40

**Comment by soc** [comment]
directories version 21 was released a few minutes ago.

**Comment by eatkins** [comment]
https://github.com/coursier/coursier/issues/1847

**Comment by eatkins** [comment]
Should be fixed by #5851
